### PR TITLE
feat: restructure how imagery configuration is stored

### PR DESCRIPTION
**Pull request by blacha**
Do not store the specific tiff files just use a s3 bucket location that can be looked up later, this greatly reduces the amount of configuration that is needed to create a basemap from imagery

### CODE REVIEW

**Comment by blacha** [comment]
This now depends on #447

File: src/base.config.ts [hunk 1]
```diff
@@ -54,53 +54,18 @@ export abstract class Updater<S extends { id: string } = { id: string }, T exten
...
-    if (oldData == null || this.showDiff(oldData, newData)) {
+    if (oldData == null || ConfigDiff.showDiff(this.db.prefix, oldData, newData, this.logger)) {
       const operation = oldData == null ? 'Insert' : 'Update';
-      this.logger.info({ type: this.db.prefix, record: newData.id }, `Change:${operation}`);
+      this.logger.info({ type: this.db.prefix, record: newData.id, commit: this.isCommit }, `Change:${operation}`);
       if (this.isCommit) {
-        if (this.db instanceof ConfigDynamoBase) await this.db.put(newData);
-        else throw new Error('Unable to commit changes to: ' + this.db.prefix);
+        if (this.db instanceof ConfigDynamoBase) {
+          console.log('Put', { newData });
```
Comment (Wentao-Kuang) [inline src/base.config.ts:62]: console.log

File: src/commands/import.ts [hunk 2]
```diff
@@ -28,10 +32,15 @@ export class CommandImport extends Command {
   async run(): Promise<void> {
     const logger = LogConfig.get();
     const { flags } = await this.parse(CommandImport);
+    if (flags.verbose) logger.level = 'trace';
 
-    for await (const fileName of fsa.list(`./config/imagery`)) this.update(fileName, flags.tag, flags.commit);
-    for await (const fileName of fsa.list(`./config/style`)) this.update(fileName, flags.tag, flags.commit);
-    for await (const fileName of fsa.list(`./config/provider`)) this.update(fileName, flags.tag, flags.commit);
+    // for await (const fileName of fsa.list(`./config/imagery`)) this.update(fileName, flags.tag, flags.commit);
```
Comment (Wentao-Kuang) [inline src/commands/import.ts:37]: Delete this comment?

File: src/commands/import.ts [hunk 3]
```diff
@@ -71,15 +85,14 @@ export class CommandImport extends Command {
       // Limit invalidations
       for (const invalidate of this.invalidates.slice(0, 10)) {
         logger.warn(`FlushCache: ${invalidate}`);
-        await invalidateCache(invalidate, flags.commit);
+        // await invalidateCache(invalidate, flags.commit);
```
Comment (Wentao-Kuang) [inline src/commands/import.ts:88]: comment out?

File: src/config.diff.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    return output;
+  }
+
+  static showDiff<T extends { id: string }>(type: string, oldData: T, newData: T, logger: LogType): boolean {
+    const changes = diff.diff(oldData, newData, (_path: string[], key: string) => IgnoredProperties.indexOf(key) >= 0);
+    if (changes) {
+      logger.info({ type, record: newData.id }, 'Changes');
+      const output = ConfigDiff.printDiff(changes);
+      console.log(output);
```
Comment (Wentao-Kuang) [inline src/config.diff.ts:39]: console.log

File: src/config/imagery.config.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      this.cache.set(uri, existing);
+    }
+    return existing;
+  }
+
+  async fetchImageryConfig(uri: string, name: string, tms: TileMatrixSet, log: LogType): Promise<ConfigImagery> {
+    log.trace({ uri }, 'FetchImagery');
+
+    // TODO is there a better way of guessing the imagery id?
+    const id = 'im_' + guessIdFromUri(uri);
```
Comment (Wentao-Kuang) [inline src/config/imagery.config.ts:31]: Maybe try to use `ulid` to validate it the id is ulid?

File: src/config/imagery.config.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    if (bounds == null) throw new Error('Failed to get bounds from URI: ' + uri);
+    const now = Date.now();
+    const output = {
+      id,
+      name,
+      createdAt: now,
+      updatedAt: now,
+      projection: tms.projection.code,
+      //   tileMatrix: tms.identifier,
```
Comment (Wentao-Kuang) [inline src/config/imagery.config.ts:75]: Comment out code?

**Review by Wentao-Kuang — APPROVED** [review]

**Review by dwsilk — APPROVED** [review]

**Review by Wentao-Kuang — APPROVED** [review]

**Pull request merged**